File: src/main/java/cz/wa/secretagent/view/renderer/model/AbstractModelDrawer.java
//package cz.wa.secretagent.view.renderer.model;
//
//import java.util.List;
//
//import org.springframework.beans.factory.annotation.Required;
//
//import cz.wa.secretagent.view.renderer.PrimitivesDrawer;
//import cz.wa.secretagent.view.texture.TextureToDraw;
//import secretAgent.world.GLModel;
//
///**
// *
// *
// * @author Ondrej Milenovsky
// */
//public abstract class AbstractModelDrawer<M extends GLModel> implements ModelDrawer<M> {
//
//    private static final long serialVersionUID = -5570243308467570348L;
//
//    protected PrimitivesDrawer primitivesDrawer;
//
//    public PrimitivesDrawer getPrimitivesDrawer() {
//        return primitivesDrawer;
//    }
//
//    @Required
//    public void setPrimitivesDrawer(PrimitivesDrawer primitivesDrawer) {
//        this.primitivesDrawer = primitivesDrawer;
//    }
//
//    protected static TextureToDraw getFrame(long currTimeMs, long durationMs, List<TextureToDraw> textures) {
//        currTimeMs = currTimeMs % durationMs;
//        int num = (int) (currTimeMs / (double) durationMs * textures.size());
//        return textures.get(num);
//    }
//
//}

File: src/main/kotlin/secretAgent/view/renderer/model/AbstractModelDrawer.kt
package secretAgent.view.renderer.model

import cz.wa.secretagent.view.renderer.PrimitivesDrawer
import cz.wa.secretagent.view.renderer.model.ModelDrawer
import cz.wa.secretagent.view.texture.TextureToDraw
import secretAgent.world.GLModel

/**
 * @author Ondrej Milenovsky
 */
abstract class AbstractModelDrawer<M : GLModel> : ModelDrawer<M> {

    lateinit var primitivesDrawer: PrimitivesDrawer

    companion object {

        private const val serialVersionUID = -5570243308467570348L

        @JvmStatic
        protected fun getFrame(currTimeMs_: Long, durationMs: Long, textures: List<TextureToDraw?>): TextureToDraw? {
            val currTimeMs = currTimeMs_ % durationMs
            val num = (currTimeMs / durationMs.toDouble() * textures.size).toInt()
            return textures[num]
        }
    }
}
